main.js

```diff
@@ -1,6 +1,59 @@
 const addButton = document.getElementById('add_button');
 const booksWrapper = document.getElementById('book-collection-wrapper');
 let populateBookSection = () => {};
+const navBtns = document.querySelectorAll('#menu li');
+
+// NAVIGATION MENU LOGIC\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+// onclick receives the index of which btn was clicked
+function HighlightSelection(index) {
+  navBtns.forEach((btn, i) => {
+    btn.classList.remove('selected');
+    if (index === i) {
+      // checks which button was clicked and adds a class
+      btn.classList.add('selected');
+    }
+  });
+}
+
+function navigate(index) {
+  const sections = ['list', 'add-new', 'contact'];
+  const bookCollection = document.getElementById('book-collection');
+  const addBookSection = document.getElementById('add-book-section');
+  const contactSection = document.getElementById('contact-section');
+  switch (sections[index]) {
+    // compares and changes the elements class accordingly to show or hide the sections
+    case 'list': {
+      bookCollection.classList.add('show');
+      addBookSection.classList.remove('show');
+      contactSection.classList.remove('show');
+      HighlightSelection(index);
+      break;
+    }
+    case 'add-new': {
+      bookCollection.classList.remove('show');
+      addBookSection.classList.add('show');
+      contactSection.classList.remove('show');
+      HighlightSelection(index);
+      break;
+    }
+    case 'contact': {
+      bookCollection.classList.remove('show');
+      addBookSection.classList.remove('show');
+      contactSection.classList.add('show');
+      HighlightSelection(index);
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+}
+
+// loop through btns adding onclick
+navBtns.forEach((btn, index) => {
+  btn.onclick = () => navigate(index);
+});
+// BOOK COLLECTION CLASS\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
 class Books {
```

Comment:
- [ ] It will be great to ensure that you can add and remove a book successfully. At the moment when I try to add a book, the following error pops up on my console, and the book is not added or displayed on the page even after I try to reload. kindly fix this problem ⬇️ 
![Screenshot 2023-08-11 at 04 40 20](https://github.com/Estete9/awesomeBooks/assets/32361963/60fcba2d-1358-4fcf-90df-95811332de32)

Reply:
I appreciate your help, please can you be more clear about how this error happened? I am not able to reproduce it